grovedb/src/query_result_type.rs

```diff
@@ -0,0 +1,83 @@
+use crate::Element;
+
+#[derive(Copy, Clone)]
+pub enum QueryResultType {
+    QueryElementResultType,
+    QueryKeyElementPairResultType,
+    QueryPathKeyElementTrioResultType,
+}
+
+pub type QueryResultItems = Vec<QueryResultItem>;
+
+pub trait GetItemResults {
```

Comment:
why to use a trait which will be never used again instead of a struct with methods?

Reply:
Because it's on Vec, so we are extending Vec with this Trait. Or at least that was my logic.

Reply:
I believe it's more than vec now, it deserves a distinct type for it (with methods and conversions)